apis/core/v1alpha1/commonobjectset_types.go

```diff
@@ -0,0 +1,152 @@
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Specifies the lifecycle state of the ObjectSet.
+type ObjectSetLifecycleState string
+
+const (
+	// "Active" is the default lifecycle state.
+	ObjectSetLifecycleStateActive ObjectSetLifecycleState = "Active"
+	// "Paused" disables reconciliation of the ObjectSet.
+	// Only Status updates will still propagated, but object changes will not be reconciled.
+	ObjectSetLifecycleStatePaused ObjectSetLifecycleState = "Paused"
+	// "Archived" disables reconciliation while also "scaling to zero",
+	// which deletes all objects that are not excluded via the pausedFor property and
+	// removes itself from the owner list of all other objects previously under management.
```

Comment:
Are the objects also deleted if there is no other owner? If so I might include that here.

Reply:
Nope. GC only kicks in, if all owners referenced by the child are deleted.
If an ObjectSet is Archived, another should have taken over, so objects are not deleted.